webapp/graphite/render/datalib.py

```diff
@@ -111,11 +114,60 @@ def fetchData(requestContext, pathExpr):
   startTime = int( epoch( requestContext['startTime'] ) )
   endTime   = int( epoch( requestContext['endTime'] ) )
 
+  def _fetchDataWrapper(node, startTime, endTime, result_queue):
+    result = node.fetch(startTime, endTime)
+    result_queue.put((node, result))
+
   def _fetchData(pathExpr,startTime, endTime, requestContext, seriesList):
+    result_queue = Queue.Queue()
```

Comment:
The existing model is asynchronous fetches to remote stores (cluster servers) and serial fetches for local (whisper files).  The asynchronous responses were then serially handled in-line with the local fetches, if any existed.

I put this patch together because I was looking at long fetch times for serially opening whisper files + carbonlink over thousands of datafiles.  This is an extreme case, but something I was looking to see if this helped out.  I do not have number yet to confirm if it does.

If this does help, then we should probably remove the async callback from remote fetches and just use threading, if threading is enabled, and not mix the two styles.